crates/test-collector/src/lib.rs

```diff
@@ -498,7 +499,18 @@ pub fn collect_tests(
         b.build()?
     };
 
+    // region: Modified code from setup_single_project_file from cairo-lang-compiler/src/project.rs
     let main_crate_id = db.intern_crate(CrateLongId::Real(SmolStr::from(crate_name)));
+    db.set_crate_root(
+        main_crate_id,
+        Some(Directory::Real(crate_root.to_path_buf())),
+    );
+
+    let module_id = ModuleId::CrateRoot(main_crate_id);
+    let file_id = db.module_main_file(module_id).unwrap();
+    db.as_files_group_mut()
+        .override_file_content(file_id, Some(Arc::new(lib_content.to_string())));
```

Comment:
Why was this code introduced?

Reply:
Updated description (cc @MaksymilianDemitraszek ):
> The approach was to manually insert the lib entrypoint content into the compiler db, that's what lib_content field in TestCompilationTarget is for.

Reply:
Will extract it to a separate function with appropriate name so the purpose is obvious for the reader